pkg/proxy/endpointslicecache_test.go

```diff
@@ -303,6 +303,41 @@ func TestEndpointInfoByServicePort(t *testing.T) {
 				},
 			},
 		},
+		"3 endpoints with deprecated topology nodeName": {
```

Comment:
Can we add one test case where the node name and the deprecated topology are populated with different values to show that node name gets precedence over the value in deprecated topology.

Reply:
nice comment, thanks!
I've added the '3 endpoints that each with different nodeName and deprecated topology hostName' case.